cartoframes/datasets.py

```diff
@@ -140,6 +141,20 @@ def _create_table_query(self, with_lonlat=None):
         create_query = '''CREATE TABLE {table_name} ({cols})'''.format(table_name=self.table_name, cols=cols)
         return create_query
 
+    def _recursive_read(self, query, retry_times=3):
+        try:
+            return self.cc.copy_client.copyto_stream(query)
+        except CartoRateLimitException as err:
+            if retry_times > 0:
+                retry_times -= 1
+                warn('Read call rate limited. Waiting {s} seconds'.format(s=err.retry_after))
+                time.sleep(err.retry_after)
+                warn('Retrying...')
+                return self._recursive_read(query, retry_times)
+            else:
+                warn('Read call was rate limited. Are you running more read queries at the same time?.')
```

Comment:
How about `Read call was rate-limited. This usually happens when there are multiple queries being read at the same time.`?

Reply:
Of course :+1: